src/include/duckdb/storage/buffer_manager.hpp

```diff
@@ -93,7 +93,15 @@ class BufferManager {
 	//! (i.e. not enough blocks could be evicted)
 	//! If the "buffer" argument is specified AND the system can find a buffer to re-use for the given allocation size
 	//! "buffer" will be made to point to the re-usable memory. Note that this is not guaranteed.
-	bool EvictBlocks(idx_t extra_memory, idx_t memory_limit, unique_ptr<FileBuffer> *buffer = nullptr);
+	//! Returns a pair. result.first indicates if eviction was successful. result.second contains the
+	//! reservation handle, which can be moved to the BlockHandle that will own the reservation.
+	std::pair<bool, TempBufferPoolReservation> EvictBlocks(idx_t extra_memory, idx_t memory_limit,
```

Comment:
Could we create a class for this and not use `pair` here? In general `pair` is not particularly descriptive - it is nicer to have variables that have descriptive names as a result.